### ISSUE TITLE
cdc - Could not update CDC description table with generation is not an error if the automatic retry works

### ISSUE THREAD

**Issue by fgelcer** [scylladb/scylla-cluster-tests#3705]
### Versions
* **SCT**: master
* **scylla**: master

### Logs
* test_id: 8cc744fb-ee25-4064-b259-41bf187b94a7
* job log: https://jenkins.scylladb.com/job/scylla-master/job/longevity/job/longevity-cdc-100gb-4h-test/240/

this test got a `DATABASE_ERROR` event with the following content:
```
2021-06-20 20:19:18.000: (DatabaseLogEvent Severity.ERROR) period_type=one-time event_id=4bf72a6a-6ded-4538-805f-11c9077c81df: type=DATABASE_ERROR regex=Exception  line_number=26259 node=Node longevity-cdc-100gb-4h-master-db-node-8cc744fb-7 [13.53.100.216 | 10.0.1.118] (seed: False)
2021-06-20T20:19:18+00:00  longevity-cdc-100gb-4h-master-db-node-8cc744fb-7 !WARNING | scylla:  [shard 0] cdc - Could not update CDC description table with generation (2021/06/20 19:42:19, eeb2241d-1831-4309-b992-0e8bff1b69bd): exceptions::unavailable_exception (Cannot achieve consistency level for cl QUORUM. Requires 2, alive 0). Will retry in the background.
2021-06-20 20:23:40.000: (DatabaseLogEvent Severity.NORMAL) period_type=one-time event_id=95a5c4ff-e19b-463f-801d-0c360651e7f6: type=BOOT regex=Starting Scylla Server line_number=27382 node=Node longevity-cdc-100gb-4h-master-db-node-8cc744fb-7 [13.53.100.216 | 10.0.1.118] (seed: False)
2021-06-20T20:23:40+00:00  longevity-cdc-100gb-4h-master-db-node-8cc744fb-7 !INFO    | systemd[1]: Starting Scylla Server...
2021-06-20 20:23:58.000: (DatabaseLogEvent Severity.ERROR) period_type=one-time event_id=4bf72a6a-6ded-4538-805f-11c9077c81df: type=DATABASE_ERROR regex=Exception  line_number=28241 node=Node longevity-cdc-100gb-4h-master-db-node-8cc744fb-7 [13.53.100.216 | 10.0.1.118] (seed: False)
2021-06-20T20:23:58+00:00  longevity-cdc-100gb-4h-master-db-node-8cc744fb-7 !WARNING | scylla:  [shard 0] cdc - Could not update CDC description table with generation (2021/06/20 19:42:19, eeb2241d-1831-4309-b992-0e8bff1b69bd): exceptions::unavailable_exception (Cannot achieve consistency level for cl QUORUM. Requires 2, alive 0). Will retry in the background.
```

but this is not really an error
as @kbr- explained, this is fine, as this happened during `MultipleHardRebootNode` nemesis, and if after that we see one of these 2 messages, it is fine:
```
cdc_log.info("Generation {}: streams description table already updated.", gen_id);
```
or
```
cdc_log.info("CDC description table successfully updated with generation {}.", gen_id);
```

in this very same test, we have after the error message this:
```
2021-06-20T20:20:18+00:00  longevity-cdc-100gb-4h-master-db-node-8cc744fb-7 !INFO    | scylla:  [shard 0] cdc - Generation (2021/06/20 19:42:19, eeb2241d-1831-4309-b992-0e8bff1b69bd): streams description table already updated.
```
and this:
```
2021-06-20T20:24:58+00:00  longevity-cdc-100gb-4h-master-db-node-8cc744fb-7 !INFO    | scylla:  [shard 0] cdc - Generation (2021/06/20 19:42:19, eeb2241d-1831-4309-b992-0e8bff1b69bd): streams description table already updated.
```

we need to think of a way that, we will ignore this error (or lower its severity) if one of the 2 "success" messages appear after that error, or either use the mechanism to lower the severity of the event with a grace period of multiples of 60 seconds (as internally CDC has a 60 seconds interval between retries).

**Comment by roydahan** [comment]
@aleksbykov can you please look into it and if needed add the correct filter?

**Comment by fruch** [comment]
This is a print with level `!WARNING` can we treat it as such ? Even that the word exception appear in it ?

**Comment by roydahan** [comment]
Yes

**Comment by fruch** [comment]
This PR would mark this line as warning:
https://github.com/scylladb/scylla-cluster-tests/pull/4293

**Comment by yarongilor** [comment]
I think the above fix and #4293 works ok, so this issue can be closed.
Got only a nemesis error of:
```
2022-05-02 22:20:02.268: (DisruptionEvent Severity.ERROR) period_type=end event_id=8d784e80-8af6-4e49-8a4b-f399e6c8e552 duration=7m14s: nemesis_name=MultipleHardRebootNode target_node=Node longevity-large-partitions-8h-5-0-db-node-a47961e7-5 [34.206.53.114 | 10.0.1.35] (seed: False) errors=After '2022-05-02T22:19:15+00:00 longevity-large-partitions-8h-5-0-db-node-a47961e7-5 ! WARNING |  [shard 0] cdc - Could not update CDC description table with generation (2022/05/02 18:15:36, 6355e121-7183-400e-9857-bc017ea4acfd): exceptions::unavailable_exception (Cannot achieve consistency level for cl QUORUM. Requires 2, alive 1). Will retry in the background.
', messages '' were not found
Traceback (most recent call last):
File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 3336, in wrapper
result = method(*args, **kwargs)
File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 633, in disrupt_multiple_hard_reboot_node
raise CdcStreamsWasNotUpdated(
sdcm.nemesis.CdcStreamsWasNotUpdated: After '2022-05-02T22:19:15+00:00 longevity-large-partitions-8h-5-0-db-node-a47961e7-5 ! WARNING |  [shard 0] cdc - Could not update CDC description table with generation (2022/05/02 18:15:36, 6355e121-7183-400e-9857-bc017ea4acfd): exceptions::unavailable_exception (Cannot achieve consistency level for cl QUORUM. Requires 2, alive 1). Will retry in the background.
', messages '' were not found
```

**Comment by fgelcer** [comment]
> I think the above fix and #4293 works ok, so this issue can be closed. Got only a nemesis error of:
> 
> ```
> 2022-05-02 22:20:02.268: (DisruptionEvent Severity.ERROR) period_type=end event_id=8d784e80-8af6-4e49-8a4b-f399e6c8e552 duration=7m14s: nemesis_name=MultipleHardRebootNode target_node=Node longevity-large-partitions-8h-5-0-db-node-a47961e7-5 [34.206.53.114 | 10.0.1.35] (seed: False) errors=After '2022-05-02T22:19:15+00:00 longevity-large-partitions-8h-5-0-db-node-a47961e7-5 ! WARNING |  [shard 0] cdc - Could not update CDC description table with generation (2022/05/02 18:15:36, 6355e121-7183-400e-9857-bc017ea4acfd): exceptions::unavailable_exception (Cannot achieve consistency level for cl QUORUM. Requires 2, alive 1). Will retry in the background.
> ', messages '' were not found
> Traceback (most recent call last):
> File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 3336, in wrapper
> result = method(*args, **kwargs)
> File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 633, in disrupt_multiple_hard_reboot_node
> raise CdcStreamsWasNotUpdated(
> sdcm.nemesis.CdcStreamsWasNotUpdated: After '2022-05-02T22:19:15+00:00 longevity-large-partitions-8h-5-0-db-node-a47961e7-5 ! WARNING |  [shard 0] cdc - Could not update CDC description table with generation (2022/05/02 18:15:36, 6355e121-7183-400e-9857-bc017ea4acfd): exceptions::unavailable_exception (Cannot achieve consistency level for cl QUORUM. Requires 2, alive 1). Will retry in the background.
> ', messages '' were not found
> ```

does it mean that, this is the message you see as the nemesis failure?
in that case, this is a problem, as this error we consider as a `WARNING`, but we are getting an error in a nemesis..
@yarongilor , can you please confirm if this is the cause of this nemesis failure, or just a symptom (message)?

**Comment by yarongilor** [comment]
@fgelcer , closing this issue. See a different related issue in: https://github.com/scylladb/scylla-cluster-tests/issues/4710
